### ISSUE TITLE
Is it correct to keep using adapter_kv_cache during training in litgpt/adapter.py?

### ISSUE THREAD

**Issue by RookieXwc** [Lightning-AI/litgpt#1287]
Is it correct to keep using `adapter_kv_cache` during training in `litgpt/adapter.py`? I think `self.adapter_wte` and `self.attn` are updated during training, so ak and av should not use `kv_cache`. Thank you very much!
But it seems that during training, `self.adapter_kv_cache` was also used.
![image](https://github.com/Lightning-AI/litgpt/assets/44500103/9c81c968-3712-4d89-acc2-40154297f2e5)

**Comment by RookieXwc** [comment]
@Andrei-Aksionov @carmocca Can you answer my doubts? I would greatly appreciate it

**Comment by Andrei-Aksionov** [comment]
Hello @RookieXwc 
Thanks for the question.

To me it also looks strange.
But I would like to have a confirmation from @carmocca that we don't miss anything.

**Comment by carmocca** [comment]
You are right. It's missing the clearing of the adapter kv cache. It could be done by overriding https://github.com/Lightning-AI/litgpt/blob/main/litgpt/model.py#L133-L136 for the adapter model

**Comment by Borda** [comment]
likely will be resolved by #2061
